infrastructure/transform/feature-data-store.tf

```diff
@@ -0,0 +1,139 @@
+  administrator_login                  = local.sql_server_features_admin_username
+  administrator_login_password         = random_password.sql_admin_password.result
+  public_network_access_enabled        = var.local_mode
+  outbound_network_restriction_enabled = true
+  tags                                 = var.tags
+}
+
+# optional firewall rule when running in local_mode
+resource "azurerm_mssql_firewall_rule" "deployer_ip_exception" {
+  count            = var.local_mode == true ? 1 : 0
+  name             = "DeployerIP"
+  server_id        = azurerm_mssql_server.sql_server_features.id
+  start_ip_address = var.deployer_ip_address
+  end_ip_address   = var.deployer_ip_address
+}
+
+# Enable Transparent Data Encryption (TDE) with a Service Managed Key
+resource "azurerm_mssql_server_transparent_data_encryption" "sql_server_features_encryption" {
+  server_id = azurerm_mssql_server.sql_server_features.id
+}
+
+# Azure SQL database, basic + small for dev
+# TODO: Rightsize for prod -> https://github.com/UCLH-Foundry/FlowEHR/issues/63
+resource "azurerm_mssql_database" "feature_database" {
+  name                 = "sql-db-features"
+  server_id            = azurerm_mssql_server.sql_server_features.id
+  collation            = "SQL_Latin1_General_CP1_CI_AS"
+  license_type         = "LicenseIncluded"
+  max_size_gb          = 2
+  sku_name             = "Basic"
+  storage_account_type = "Local"
+  zone_redundant       = false
+  tags                 = var.tags
+}
+
+# AAD App + SPN for Databricks -> SQL Access
+resource "azuread_application" "flowehr_databricks_sql" {
+  display_name = "FlowEHR-Databricks-SQL-${var.naming_suffix}"
+}
+resource "azuread_service_principal" "flowehr_databricks_sql" {
+  application_id = azuread_application.flowehr_databricks_sql.application_id
+}
+resource "azuread_service_principal_password" "flowehr_databricks_sql" {
+  service_principal_id = azuread_service_principal.flowehr_databricks_sql.object_id
+}
+
+# Push secrets to KV
+resource "azurerm_key_vault_secret" "sql_server_features_admin_username" {
+  name         = "sql-features-admin-username"
+  value        = local.sql_server_features_admin_username
+  key_vault_id = var.core_kv_id
+}
+resource "azurerm_key_vault_secret" "sql_server_features_admin_password" {
+  name         = "sql-features-admin-password"
+  value        = random_password.sql_admin_password.result
+  key_vault_id = var.core_kv_id
+}
+resource "azurerm_key_vault_secret" "flowehr_databricks_sql_spn_app_id" {
+  name         = "flowehr-dbks-sql-spn-app-id"
+  value        = azuread_service_principal.flowehr_databricks_sql.application_id
+  key_vault_id = var.core_kv_id
+}
+resource "azurerm_key_vault_secret" "flowehr_databricks_sql_spn_app_secret" {
+  name         = "flowehr-dbks-sql-spn-app-secret"
+  value        = azuread_service_principal_password.flowehr_databricks_sql.value
+  key_vault_id = var.core_kv_id
+}
+
+# Push SPN details to databricks secrets (backed by KV)
```

Comment:
Confused about this comment. Is it still backed by KV? Or is it a built-in secret scope?

Reply:
bad comment leftover from WIP! good spot! will remove!